web/modules/custom/collection/src/Plugin/Block/JoinCollectionBlock.php

```diff
@@ -84,12 +80,13 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function build() {
-    if (empty($this->collection)) {
```

Comment:
If the context is missing, then the `->getContextValue() will be called on null and an exception will be fired. I think we should check if the value is empty since we throw our own exception at the next line based on the existence of the context here.